## Pruebas inciales para vision de Azure, implementacion de reconocimiento de imagenes 

```Python
# Instalacion necesaria previa
pip install azure-cognitiveservices-vision-computervision 
pip install azure-ai-textanalytics

In [ ]:
from azure.cognitiveservices.vision.computervision import ComputerVisionClient
from azure.cognitiveservices.vision.computervision.models import VisualFeatureTypes
from msrest.authentication import CognitiveServicesCredentials



client = ComputerVisionClient(
    endpoint="https://proyectoazuredll.cognitiveservices.azure.com/",
    credentials=credentials
)

In [11]:
url = "https://upload.wikimedia.org/wikipedia/commons/3/3a/Cat03.jpg"

image_analysis = client.analyze_image(url,visual_features=[VisualFeatureTypes.tags], language="es")

for tag in image_analysis.tags:
    print(tag)

{'additional_properties': {}, 'name': 'animal', 'confidence': 0.9991010427474976, 'hint': None}
{'additional_properties': {}, 'name': 'gato', 'confidence': 0.9985039830207825, 'hint': None}
{'additional_properties': {}, 'name': 'mamíferos', 'confidence': 0.9980359077453613, 'hint': None}
{'additional_properties': {}, 'name': 'bigotes', 'confidence': 0.9781772494316101, 'hint': None}
{'additional_properties': {}, 'name': 'gatos de tamaño pequeño a mediano', 'confidence': 0.9777382612228394, 'hint': None}
{'additional_properties': {}, 'name': 'gato doméstico', 'confidence': 0.9691081047058105, 'hint': None}
{'additional_properties': {}, 'name': 'félidos', 'confidence': 0.9345883131027222, 'hint': None}
{'additional_properties': {}, 'name': 'mau árabe', 'confidence': 0.8746515512466431, 'hint': None}
{'additional_properties': {}, 'name': 'gato asiático', 'confidence': 0.8502976894378662, 'hint': None}
{'additional_properties': {}, 'name': 'naranja', 'confidence': 0.769542396068573, 'hint'

### Etiquetas que han salido en la prueba
Con la url https://upload.wikimedia.org/wikipedia/commons/3/3a/Cat03.jpg he detectado las siguientes etiquetas:

* {'additional_properties': {}, 'name': 'animal', 'confidence': 0.999100923538208, 'hint': None}
* {'additional_properties': {}, 'name': 'cat', 'confidence': 0.9985039234161377, 'hint': None}
* {'additional_properties': {}, 'name': 'mammal', 'confidence': 0.9980359077453613, 'hint': None}
* {'additional_properties': {}, 'name': 'whiskers', 'confidence': 0.9781772494316101, 'hint': None}
* {'additional_properties': {}, 'name': 'small to medium-sized cats', 'confidence': 0.9777383804321289, 'hint': None}
* {'additional_properties': {}, 'name': 'domestic cat', 'confidence': 0.9691082239151001, 'hint': None}
* {'additional_properties': {}, 'name': 'felidae', 'confidence': 0.9345883131027222, 'hint': None}
* {'additional_properties': {}, 'name': 'arabian mau', 'confidence': 0.8746517896652222, 'hint': None}
* {'additional_properties': {}, 'name': 'malayan cat', 'confidence': 0.8502976894378662, 'hint': None}
* {'additional_properties': {}, 'name': 'orange', 'confidence': 0.7695426940917969, 'hint': None}
* {'additional_properties': {}, 'name': 'ground', 'confidence': 0.6549572944641113, 'hint': None}
* {'additional_properties': {}, 'name': 'looking', 'confidence': 0.5822522640228271, 'hint': None}
* {'additional_properties': {}, 'name': 'outdoor', 'confidence': 0.5709481835365295, 'hint': None}

El confidence es la confianza que tiene el modelo de que la etiqueta es correecta.

In [8]:
models = client.list_models()

for x in models.models_property:
    print(x)

{'additional_properties': {}, 'name': 'celebrities', 'categories': ['people_', '人_', 'pessoas_', 'gente_']}
{'additional_properties': {}, 'name': 'landmarks', 'categories': ['outdoor_', '户外_', '屋外_', 'aoarlivre_', 'alairelibre_', 'building_', '建筑_', '建物_', 'edifício_']}


In [21]:
domain = "landmarks"
url = "https://images.pexels.com/photos/338515/pexels-photo-338515.jpeg"
language = "es"

analysis = client.analyze_image_by_domain(domain, url, language)

for landmark in analysis.result["landmarks"]:
    print(landmark["name"])
    print(landmark["confidence"])

Torre Eiffel
0.971265435218811


### Prueba con la lista de modelos
Con celebrities no he podido usarlo por falta de permisos para poder detectar caras.

In [14]:
url = "https://www.vinccihoteles.com/media/uploads/cms_apps/imagenes/la-pedrera-barcelona.jpg?q=pr:sharp/rs:fill/w:900/h:500/g:ce/f:jpg"
language = "es"
max_descriptions = 3

analysis = client.describe_image(url, max_descriptions, language)

for caption in analysis.captions:
    print(caption.text)
    print(caption.confidence)

ciudad con edificios altos con Casa Milà de fondo
0.7334685605131565
Casa Milà de fondo
0.704811825024861
vista de Casa Milà
0.6901420789523418


### Pruebas de analisis por dominio y descripción de la imágen 
Con la url https://www.vinccihoteles.com/media/uploads/cms_apps/imagenes/la-pedrera-barcelona.jpg?q=pr:sharp/rs:fill/w:900/h:500/g:ce/f:jpg he detectado las siguientes descripciones:

* {'text': 'a large building with a clock tower', 'confidence': 0.971, 'hint': None}
* {'text': 'a large building with a clock tower and a street in front of it', 'confidence': 0.971, 'hint': None}
* {'text': 'a large building with a clock tower and a street in front of it', 'confidence': 0.971, 'hint': None}